src/python/pants/base/specs.py

```diff
@@ -528,3 +540,68 @@ def arguments_provided_description(self) -> str | None:
         if len(specs_descriptions) == 2:
             return " and ".join(specs_descriptions) + " arguments"
         return ", ".join(specs_descriptions[:-1]) + f", and {specs_descriptions[-1]} arguments"
+
+
+# -----------------------------------------------------------------------------------------------
+# Plugin API for injecting additional paths in addition to those found using BUILD file globs for
+# the current specs.
+# -----------------------------------------------------------------------------------------------
+
+T = TypeVar("T", bound="AdditionalSpecPathsRequest")
+
+
+@union
```

Comment:
I'm not sure that adding a `@union` here is a good idea: synthetic targets are a built in concept that we don't necessarily want users to need to implement themselves.

Reply:
I didn't at first, but that leaves plugins that registers synthetic targets per-directory in the dark unless their directory also has a BUILD file in it, with no easy way out.

Reply:
That sounds like an issue with the synthetic targets API though... it would be error prone to need to install two unions to make synthetic targets work.

Reply:
I'll re-work the synthetic targets API so that we can remove this `@union`.

Reply:
Thanks a lot for all of work on that. It's very appreciated.

Reply:
Well, I only "moved" the `@union` so it's more of an implementation detail now for synthetic targets, and not something that can be abused willy-nilly to add additional spec paths.